next-cloudinary/src/components/CldVideoPlayer/CldVideoPlayer.tsx

```diff
@@ -161,6 +161,16 @@ const CldVideoPlayer = (props: CldVideoPlayerProps) => {
     }
   }
 
+  useEffect(() => {
+    handleOnLoad();
+
+    return () => {
```

Comment:
it looks like this could use a `destroy`-like method, right? did you intentionally not add one here?

the video player has the `dispose` method which could apply:

https://cloudinary.com/documentation/video_player_api_reference#dispose

what you have works, but i wonder / worry if it's not being properly removed from memory

Reply:
The problem is that the `handleOnLoad` function only runs on initial page load. When you navigate to another page and back, it does not run again. But this behaviour does not happen with the `CldUploadWidget` component, every time the component mounts, the function runs. Any ideas why that may be happening ? @colbyfayock

Reply:
this is an interesting one, it looks like the difference is that the upload widget includes an ID of:
```
id={`cloudinary-uploadwidget-${Math.floor(Math.random() * 100)}`}
```
When appending that to the player, while commenting out `handleOnLoad` inside of `useEffect`, it works a little more consistently:
```
id={`cloudinary-videoplayer-${playerId}-${Math.floor(Math.random() * 100)}`}
``
however i still consistently get about 1 player that doesn't properly load

I would be more interested in learning why 1 player doesnt load - is the script not available in time? 

if you add a console log inside of `handleOnLoad`, something simple, you'll see in the devtools how many times it first, and it fires only the number of videos that appear, so if several dont load, you dont get 20 console logs

<img width="164" alt="image" src="https://github.com/colbyfayock/next-cloudinary/assets/1045274/37c9b7fc-4475-4968-9181-ce75975054bd">

it's as if Next.js or React can't handle that many Script tags all referencing the same thing at once?

Reply:
Ah I see, I just tested now with this `id={`cloudinary-videoplayer-${playerId}-${Math.floor(Math.random() * 100)}`}`as the id prop in the script tag and it works. I wonder why it was having issues without the `Math.random` as part of the id. 

You also need to pass a unique `id` prop for each video when using the component. With this, each video loads fine I believe.